### ISSUE TITLE
release on conda-forge - it's now easy!

### ISSUE THREAD

**Issue by rsignell-usgs** [opendatacube/odc-tools#469]
The odc-tools are great and it would be great to allow simple install for users from conda-forge.  

Releasing on conda-forge used to be significant extra work, but if you release on pypi, now  [grayskull](https://github.com/conda-incubator/grayskull) has made generating the recipe a one-liner:
```
grayskull pypi odc-algo
```
Once it's on conda-forge, the conda-forge bot will update the version and the dependencies automagically when pypi is updated.     If the package metadata is correct upstream then there is nothing for the conda-forge folks to do -- they just merge.  

@ocefpaf, please correct me if I got any of this wrong.

**Comment by Kirill888** [comment]
@rsignell-usgs `odc-algo` specifically is on conda, just not `conda-forge`. Issue is not so much with setting up conda packaging, but rather `conda-forge` distribution itself, which takes time for package to be approved and all that. Also a lot of these packages still lack stability, we had a major refactor of package boundaries a few months back for example, so adding them to `conda-forge` might be a bit premature.


```
conda install -c kirill-odc odc-algo
mamba install -c kirill-odc odc-algo
```

https://anaconda.org/kirill-odc/odc-algo

**Comment by rsignell-usgs** [comment]
@Kirill888, I think it's fine to use conda-forge in a "release early/release often" mode, it's easy now with grayskull.  

We went ahead and did it for `odc-algo`, as I want to share https://github.com/Element84/geo-notebooks with colleagues and I've been telling them to just use conda-forge to avoid issues.

**Comment by Kirill888** [comment]
> We went ahead and did it for `odc-algo`, ...

@rsignell-usgs can you elaborate? Did you make a package for `odc-algo` or did you publish it to `conda-forge`? Up-to-date conda package for `odc-algo` is available from my conda channel `kirill-odc` as I mentioned in the post earlier. I will update installation instructions for the conda section to mention that fact.

While I agree that sharing early is good, `conda-forge` is a bit of a special case here. Unlike `pip` or custom `conda` channel it requires human vetting as part of the process for package to be accepted, and I agree with that model. But what that means is that I'll need to get all the dependencies of `odc-algo` published first, and that also require human vetting, and these dependencies are still quite volatile and also not obviously useful on their own.

I agree, it's probably time to carefully analyze remaining dependencies of `odc-algo` and see if we can publish on `conda-forge`, I totally understand that `pip` installation for scientific and especially spatial workflows is just too cumbersome compared to `conda`, that's why readme includes detailed instructions how to install `odc-tools` `pip` packages within a conda environment.

**Comment by rsignell-usgs** [comment]
@Kirill888 , @ocefpaf helped me on this, and will respond with some more info.

**Comment by ocefpaf** [comment]
> @rsignell-usgs can you elaborate? Did you make a package for `odc-algo` or did you publish it to `conda-forge`?

In conda-forge lingo "making a package" is to package existing software and publishing it. So it is the latter and not the former.

> Up-to-date conda package for `odc-algo` is available from my conda channel `kirill-odc` as I mentioned in the post earlier. I will update installation instructions for the conda section to mention that fact.

Thanks. We discourage the use of multiple channels though to avoid ABI incompatibilities. We also try to make it easy to publish on community channels (conda-forge and bioconda). That usually to help folks make their packages conda installable to a larger community.

With that said, there are rare cases where you may want your own channel like security, experimental packages, etc. I would like to note that if a package is published at PyPI, like odc-algo, it is good enough yo go into conda-forge too (or any other packing platform).

> While I agree that sharing early is good, `conda-forge` is a bit of a special case here. Unlike `pip` or custom `conda` channel it requires human vetting as part of the process for package to be accepted, and I agree with that model. But what that means is that I'll need to get all the dependencies of `odc-algo` published first, and that also require human vetting, and these dependencies are still quite volatile and also not obviously useful on their own.

Most of the dependencies are already there and that is a requirement for PyPI or a custom channel too. The vetting only happen once, at staged-recipes, and we don't see that any different from the automatic checks one have to do to check their package for PyPI. We only check a bit further to ensure the package uses all the advantages conda/conda-forge can provide. Also, most of the times it is not the author sending the package, if it is the author we don't really have a lot to check and the PR is accepted much quicker. Almost like publishing at PyPI or your own channel.

> I agree, it's probably time to carefully analyze remaining dependencies of `odc-algo` and see if we can publish on `conda-forge`, I totally understand that `pip` installation for scientific and especially spatial workflows is just too cumbersome compared to `conda`, that's why readme includes detailed instructions how to install `odc-tools` `pip` packages within a conda environment.

When not using a community channel you are forcing users to [mix channels](https://conda-forge.org/docs/user/tipsandtricks.html#using-multiple-channels), like defaults and/or conda-forge, with yours. That can lead to many headaches and make the installation even more cumbersome. I would recommend an approach that uses conda+pip instead of another channel. Or, better yet, submit your package to conda-forge.

PS: I am a bit biased here b/c I deal with channel mixing problems and user confusion every day. I can add the packages you published on PyPI if you want. It is really quick and painless now. We have a tool that creates the recipe and, once the package is accepted, a bot will update it for you. You can even add automerge if you don't want to bother reviewing the bot's PR. The only downside automerge is that any bad metadata from upstream will be propagated to the package. B/c you are the author you will probably know if automerge is safe or not for you.

**Comment by rsignell-usgs** [comment]
TL;DR:  the user community would be well served with ODC tools released early and often on conda-forge, and it wouldn't put significant extra effort on any humans!   😄

**Comment by Kirill888** [comment]
Thanks for detailed reply @ocefpaf. I see your point about mixed channels, in this case all the packages are python only, but of course you can't know if it will stay that way. The plan is of course to publish on `conda-forge` like we did with `datacube` and `odc-stac`, it's just we had a rather hairy dependency graph in the recent past, but that got mostly cleaned up by now and is likely to stay unchanged. So probably good time to publish it to `conda-forge`.

**Comment by Kirill888** [comment]
So I see @ocefpaf done this already, thanks

https://github.com/conda-forge/odc-algo-feedstock
